eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -110,165 +99,194 @@ def reduceJWST(eventlabel):
     exec('from . import ' + meta.inst + ' as inst', globals())
     reload(inst)
 
-    # Open new log file
-    meta.logname = './' + meta.workdir + '/S3_' + meta.eventlabel + ".log"
-    log = logedit.Logedit(meta.logname)
-    log.writelog("\nStarting Stage 3 Reduction")
-
-    # Create list of file segments
-    meta = util.readfiles(meta)
-    num_data_files = len(meta.segment_list)
-    log.writelog(f'\nFound {num_data_files} data file(s) ending in {meta.suffix}.fits')
-
-    stdspec = np.array([])
-    # Loop over each segment
-    # Only reduce the last segment/file if testing_S3 is set to True in ecf
-    if meta.testing_S3:
-        istart = num_data_files - 1
+    # check for range of spectral apertures
+    if isinstance(meta.spec_hw, list):
+        spec_hw_range = range(meta.spec_hw[0], meta.spec_hw[1], meta.spec_hw[2])
```

Comment:
Should we be inclusive in our range?  In other words, if someone sets spec_hw = [10,20,2], should the last value in the range be 18 or 20?  I would think 20, which would mean we need `range(meta.spec_hw[0], meta.spec_hw[1]+meta.spec_hw[2], meta.spec_hw[2])`

Reply:
Valid, I assumed the user would understand python ranges aren't inclusive (since I'm used to it) but it's much better to do things so that they make sense logically instead of assume the user knows the code well! I'll change it